tests/unit/helpers/users.test.js

```diff
@@ -143,6 +180,111 @@ describe('formatQueryForUsersList', () => {
       [{ query: 'find', args: [{ company: { $in: [companyId] } }, { user: 1 }] }, { query: 'lean' }]
     );
     sinon.assert.notCalled(find);
+    sinon.assert.notCalled(findOneCompanyHolding);
+    sinon.assert.notCalled(findUserHolding);
+  });
+
+  it('should return params with company (without includesHoldingAdmins)', async () => {
+    const users = [{ _id: new ObjectId(), user: new ObjectId() }];
+    const query = { company: companyId };
+
+    findUserCompany.returns(SinonMongoose.stubChainedQueries(users, ['lean']));
+
+    const result = await UsersHelper.formatQueryForUsersList(query);
+    expect(result).toEqual({ _id: { $in: users.map(u => u.user) } });
+
+    SinonMongoose.calledOnceWithExactly(
+      findUserCompany,
+      [
+        {
+          query: 'find',
+          args: [
+            {
+              company: companyId,
+              $or: [{ endDate: { $gt: '2022-12-21T16:00:00.000Z' } }, { endDate: { $exists: false } }],
+            },
+            { user: 1 },
+          ],
+        },
+        { query: 'lean' },
+      ]
+    );
+    sinon.assert.notCalled(find);
+    sinon.assert.notCalled(findOneCompanyHolding);
+    sinon.assert.notCalled(findUserHolding);
+  });
+
+  it('should return params with company (with includesHoldingAdmins and holding exists)', async () => {
+    const users = [{ _id: new ObjectId(), user: new ObjectId() }];
+    const holdingId = new ObjectId();
+    const companyHolding = { _id: new ObjectId(), holding: holdingId };
+    const userHoldings = [{ _id: new ObjectId(), user: new ObjectId() }];
+    const query = { company: companyId, includesHoldingAdmins: true };
+
+    findUserCompany.returns(SinonMongoose.stubChainedQueries(users, ['lean']));
+    findOneCompanyHolding.returns(SinonMongoose.stubChainedQueries(companyHolding, ['lean']));
+    findUserHolding.returns(SinonMongoose.stubChainedQueries(userHoldings, ['lean']));
+    const result = await UsersHelper.formatQueryForUsersList(query);
+    expect(result).toEqual({ _id: { $in: users.map(u => u.user) } });
```

Comment:
il manque ce que renvoie l'appel `UserHolding.find` dans ton retour non ?

Reply:
je me demande si il n'y a pas une valeur qui est changé par reference ici ce qui rend la lecture du test incohérente

Quand on lit ton test, le retour doit etre `{ _id: { $in: [...users, ...userHoldings].map(u => u.user) } }`

Or la ton test passe alors que tu dis juste ``{ _id: { $in: users.map(u => u.user) } }`

Je pense que ca vient du `users.push(...userHoldings);` dans ton code qui du coup vient modifier la variable de ton test ici

Reply:
tu peux d'ailleurs le voir en ajoutant  : 
- `console.log('test users avant', users)` avant `const result = await UsersHelper.formatQueryForUsersList(query);`
- et `console.log('test users après', users)` juste après

les 2 valeurs ne sont pas les mêmes